builtin/providers/google/resource_compute_vpn_gateway.go

```diff
@@ -0,0 +1,121 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func resourceComputeVpnGateway() *schema.Resource {
+	return &schema.Resource{
+		// Unfortunately, the VPNGatewayService does not support update
+		// operations. This is why everything is marked forcenew
+		Create: resourceComputeVpnGatewayCreate,
+		Read:	resourceComputeVpnGatewayRead,
+		Delete: resourceComputeVpnGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:	  schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"network": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"self_link": &schema.Schema{
+				Type:	  schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceComputeVpnGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	name := d.Get("name").(string)
+	network := d.Get("network").(string)
+	description := d.Get("description").(string)
+	region := d.Get("region").(string)
```

Comment:
If you started using `getOptionalRegion` above, why not use it in these two new resources as well?
Even though it doesn't feel as elegant as if we were able to use it directly in the schema via `DefaultFunc`, I think it's a nice feature and almost expected behaviour - less required fields that can be inherited from the provider.